screens/NativeModule.tsx

```diff
@@ -1,29 +1,22 @@
 import React, { useState, useEffect, useCallback } from 'react';
-import { StyleSheet, Text, View } from 'react-native';
 import { colors } from '../style/styleVariables';
 import ContainerFull from '../components/ContainerFull';
 import Header from '../components/Header';
 import { Camera } from 'expo-camera';
 import CameraComponent from '../components/CameraComponent';
 import MorseForm from '../components/MorseForm';
-import { translateToMorse } from '../morseMessenger/morseFunctions';
-import { DictOptions } from '../morseMessenger/morseDictionary';
-
-const TORCH_ON = 'torch';
-const TORCH_OFF = 'off';
-const PAUSE = 1000;
-const FLASH_PAUSE = 400;
-const FLASH_SHORT = 100;
-const FLASH_LONG = 700;
+import useFlash from '../morseMessenger/flashHook';
 
 export default function NativeModule() {
   const [hasPermission, setHasPermission] = useState<boolean | null>(null);
-  const [flashMode, setFlashMode] = useState(TORCH_OFF);
-  const [buttonDisabled, setButtonDisabled] = useState(false);
+  const [flashMode, buttonDisabled, startMorseForInput] = useFlash();
 
-  const buttonCallback = useCallback(input => {
-    asyncFlash(input);
-  }, []);
+  const buttonCallback = useCallback(
+    input => {
+      startMorseForInput(input);
+    },
+    [startMorseForInput]
```

Comment:
So this is why hooks are complex if you want to do them correctly 😄 but this `useCallback` doesn't have sense in the current logic.

In `useFlash` you define `startMorseForInput` as:
```ts
async function startMorseForInput(input:string) {
...
```
This means that this function will be recreated in every `useFlash` call.

`useCallback` is a function which makes sure that the function inside won't be recreated unless something in the dependency array (`startMorseForInput`) has changed.

So `startMorseForInput` will be always recreated now, so `useCallback` does not give as any gain as it will recreate the function anyway. Quite the contrary -> we will always check if `startMorseForInput` has changed even though we always know that it has -> such checks can sometimes slow the application down.

In simple cases like this screen, where we don't really care about performance that much, you could just pass `startMorseForInput` without `useCallback` as it doesn't hurt us to recreate a few functions in every render.

Unfortunately, in more complex projects it will 😢 The correct approach here is to put `useCallback` around `startMorseForInput` inside `useFlash` hook. You can even call it `startMorseForInputCallback` then.